Ewl 3276 table

rpama opened the pull request:
<!-- NOTE: Please just put "N/A" for any section below that isn't applicable to the work you've done, do not omit entirely. -->

## JIRA Ticket(s)
- [EWL-3276: Table](https://issues.ama-assn.org/browse/EWL-3276)


## Description:
Adds a new pattern (atom) for Caption
Adds a new pattern (molecule) for Table

## To Test:
- [x] Go to Atoms > Text > Caption
- [x] Compare pattern to comp
- [x] Go to Molecules > Blocks > Table block
- [ ] Compare pattern to comp


## Automated Test:
N/A


## Relevant Screenshots/GIFs:
Use something like [Skitch](https://evernote.com/skitch/) or [GIPHY Capture](https://giphy.com/apps/giphycapture) to capture images/gifs to demonstrate behaviors.


## Remaining Tasks:
Could make atoms out of the table header and table, but since these elements don't ever stand alone outside of a table, I'm not sure that it makes sense to do so. Thoughts?


## Additional Notes:
No.


---

[Pull Request Guidelines](https://github.com/palantirnet/development_documentation/blob/master/guidelines/pr_review.md)

deadarm commented on styleguide/source/_patterns/01-molecules/06-components/15-table/table.twig:
```diff
@@ -0,0 +1,50 @@
+      </tr>
+    </thead>
+    <tbody>
+      <tr class="odd">
```
Suggestion to use CSS even and odd pseudo classes instead of even and odd class in html

deadarm commented on styleguide/source/_patterns/01-molecules/06-components/15-table/table.twig:
```diff
@@ -0,0 +1,50 @@
+
+  {% include 'atoms-h5' with { 'content': 'Lorem Ipsum Dolor Amet Consectetur Adipiscing', 'class': 'table-header_title' } %}
+
+  <table>
```
Do we want to add a class here of table instead of applying styling directly to html element?

deadarm commented on styleguide/source/_patterns/01-molecules/06-components/15-table/_table.scss:
```diff
@@ -0,0 +1,58 @@
+  }
+
+  thead tr {
+    background-color: #e6e6e6;
```
I did a quick search and we are using #e6e6e6 in other places as a background color. It would be good to convert it to a variable.

deadarm commented on styleguide/source/assets/css/scss/generic/_variables.scss:
```diff
@@ -44,6 +44,9 @@ $blue-lighter: #d9f2fc;
 
+// Custom color variables
+
+$rule:			$gray-8;
```
See note above about rule styling.

rpama replied:
So, separate from your point about horizontal-rules, I added the $rule variable because of color variable usage I noticed in Corp when stripping out all the Chroma stuff. This is probably part of a bigger rethinking of how we organize colors overall, and I really haven't figured that all out yet.

deadarm replied:
Ok that makes sense, thanks for context!

deadarm commented on styleguide/source/_patterns/01-molecules/06-components/15-table/_table.scss:
```diff
@@ -0,0 +1,58 @@
+    @include grid__unit--cols(6);
+  }
+
+  border-top: 1px solid $rule;
```
In _global-classes.scss we have a  `.rule_horizontal {
  border-top: 1px $gray-8 solid;
}` class set up, so it would be ideal to combine these with a mixin, or note the reasons why we need both with inline comments.

deadarm commented on styleguide/source/_patterns/01-molecules/06-components/15-table/_table.scss:
```diff
@@ -0,0 +1,58 @@
+    background-color: #e6e6e6;
+  }
+
+  tr.even {
```
Suggestion to use CSS even pseudo class.

deadarm commented on styleguide/source/_patterns/00-atoms/02-text/14-caption/caption.twig:
```diff
@@ -0,0 +1 @@
+<p class="caption {{ class }}">{{ content|default('Bacon ipsum dolor amet minim biltong culpa mollit brisket labore porchetta. Turkey porchetta magna, do doner burgdoggen aute excepteur nisi sint pancetta ipsum ea. Frankfurter burgdoggen pastrami eiusmod in.') }}</p>
```
Suggestion to convert to bacon ipsum across styleguide, or use lorem ipsum here for consistency.

deadarm commented on styleguide/source/_patterns/01-molecules/06-components/15-table/_table.scss:
```diff
@@ -0,0 +1,58 @@
+/**
+ * @file accordion.scss
+ *
+ * Copyright 2017 Palantir.net, Inc.
```
@rpama Follow up that only Palantiri need to add the Palantir copyright notice when creating a file, and there is an open question on how to improve the process with a blanket statement.

rpama commented:
- Used css pseudo-class for odd/even tr -- was following what Drupal actually spit out when it made tables, but agreed those classes aren't needed. 
 - Added color variable and updated all instances where that gray has been used.
 - Cleaned up the containing structure a bit; re-categorized this as a block
 - Removed Palantir copyright from files touched in my commits. 
 - Made my bacon ipsum less conspicuous, but it's placeholder text so I'm not going to worry overmuch about consistency.

deadarm commented:
Looks good to me!
